src/main/java/example/movies/executor/CypherExecutor.java

```diff
@@ -1,11 +1,55 @@
 package example.movies.executor;
 
+import org.neo4j.driver.AuthTokens;
+import org.neo4j.driver.Config;
+import org.neo4j.driver.Driver;
+import org.neo4j.driver.GraphDatabase;
+import org.neo4j.driver.Session;
+import org.neo4j.driver.SessionConfig;
+import org.neo4j.driver.Value;
+
 import java.util.List;
 import java.util.Map;
 
 /**
  * @author Michael Hunger @since 22.10.13
  */
-public interface CypherExecutor {
-    List<Map<String, Object>> query(String statement, Map<String,Object> params);
+public class CypherExecutor {
+
+    private final Driver driver;
+
+    private final String database;
+
+    public CypherExecutor(String url, String username, String password, String database) {
+        this.driver = GraphDatabase.driver(
+                String.format("neo4j://%s", url),
+                AuthTokens.basic(username, password),
+                Config.builder().withoutEncryption().build());
+        this.database = database;
+    }
+
+    public List<Map<String, Object>> query(String query, Map<String, Object> params) {
```

Comment:
if we update this file, we can also simplify the whole project, which is now a bit overengineered :)

Reply:
I would like to do that in a subsequent PR, once the CI is fully in place, if that's OK

Reply:
changed my mind, done.